Mathport/Syntax/Translate/Basic.lean

```diff
@@ -1149,10 +1259,10 @@ def trField : Field → Array Syntax → M (Array Syntax)
         `(Parser.Command.structSimpleBinder| $(ns[0]):ident $[$im]? $sig:optDeclSig $[$dflt]?)
       else
         `(Parser.Command.structExplicitBinder| ($ns* $[$im]? $sig:optDeclSig $[$dflt]?))
-  | Field.notation _, out => warn! "unsupported: (notation) in structure"
+  | Field.notation _ => warn! "unsupported: (notation) in structure"
 
 def trFields (flds : Array (Spanned Field)) : M Syntax := do
-  let flds ← flds.foldlM (fun out fld => trField fld.kind out) #[]
+  let flds ← flds.concatMapM trField
```

Comment:
sad face for removing my optimization

Reply:
Yeah sorry, this is pretty hard to read though and doesn't work well with operations that modify the context.  That is, `withRef` etc. can leak across the iteration.  The optimization seems somewhat premature though since we're talking about arrays with <10 elements or so here.